lib/gui/modules/analytics.js

```diff
@@ -120,17 +142,22 @@ analytics.service('AnalyticsService', function($log, $window, $mixpanel, Setting
    */
   this.logEvent = (message, data) => {
 
-    if (SettingsModel.get('errorReporting') && isRunningInAsar()) {
+    // Mixpanel expects a flat object
+    const flattenedData = data ? flatten(data, {
+      delimiter: ' ',
+      safe: true
+    }) : undefined;
 
-      // Clone data before passing it to `mixpanel.track`
-      // since this function mutates the object adding
-      // some custom private Mixpanel properties.
-      $mixpanel.track(message, _.clone(data));
+    const startCaseFlattenedData = flattenedData ? _.mapKeys(flattenedData, (value, key) => {
+      return _.startCase(key);
+    }) : undefined;
 
+    if (SettingsModel.get('errorReporting') && isRunningInAsar()) {
+      $mixpanel.track(message, _.mapKeys(flattenedData, _.startCase));
```

Comment:
Why are we duplicating `_.mapKeys` here? Maybe I'm missing something, but I think you can simply pass `startCaseFlattenedData` here.